### PR TITLE
POC: rewrite gulpfile in ts using slurpy tasks

### PR DESCRIPTION
**Pull request by stevenpollack**
I ported over the typescript-template gulpfile's logic to a dedicated repo: https://github.com/proximacapital/slurpy and in the process realized that gulp is capable to use ts-node to transpile gulpfile.ts with no real effort on our part:
```
npx gulp --tasks
```
pull this branch down, install it and run the above command.

### CODE REVIEW

**Comment by OliverNChalk** [comment]
Nice! I like the promise of DRYing our Gulpfiles.

Two things so far:
 1. Why `slurpy` and not a name that is more understandable like `gulp-template`, `gulpfile`, etc? Do you envision `slurpy` going beyond `gulp` DRYing and thus the name leaves that door open?
 2. Can you demo how one would extend the `gulp build` task, say if we needed it to generate some `typechain` bindings? I assume its possible to add steps before and after but not open up the tasks (the whole point of the abstraction)?

**Comment by stevenpollack** [comment]
> 1. Why `slurpy` and not a name that is more understandable like `gulp-template`, `gulpfile`, etc? Do you envision `slurpy` going beyond `gulp` DRYing and thus the name leaves that door open?

Well there's a right and wrong answer here: Steven's Latest Undertaking to Repair Proxima's developer experIence -- SLURPI.

Honestly, I'm not precious about the name. It was more of a throwaway/deep cut for the peanut gallery.

> 2. Can you demo how one would extend the `gulp build` task, say if we needed it to generate some `typechain` bindings? I assume its possible to add steps before and after but not open up the tasks (the whole point of the abstraction)?

So, locally, you could have some sort of gulp task sitting in `Src/Gulp/GenerateTypechains.ts`, and then in `Gulpfile.ts`
```ts
import { Build } from ''@proximacapital/slurpy"
import gulp from ''gulp"
import { GenerateTypechains } from "@Gulp/GenerateTypechains"
// snip
gulp.task("build", gulp.series(GenerateTypechains, Build));
```

**Comment by OliverNChalk** [comment]
The task would be [quite simple](https://github.com/proximacapital/nav-striker/blob/9203fa55e7422f91250661f3c07e8821ec9f853d/Gulpfile.js#L55), so I'd try to in-line it in the local gulpfile. That'd be fine too?

**Comment by stevenpollack** [comment]
For sure! The location of the particular TaskFunction is largely a matter
of style.

File: package.json [hunk 1]
```diff
@@ -53,6 +53,8 @@
   },
   "devDependencies": {
     "@proxima-oss/eslint-config": "5.0.0",
+    "@proximacapital/slurpy": "0.0.1-3",
```
Comment (OliverNChalk) [inline package.json]: can we rename `slurpy` to `gulp-base`? I also considered `gulp-template` but this feels more like a base to extend from than a "template"

**Review by OliverNChalk — CHANGES_REQUESTED** [review]
I think the changes are good, just want to change the name of the upstream dependency to be self-documenting

**Comment by stevenpollack** [comment]
I don't really feel like "self-documenting" repo names are actually a
thing. I'd suspect it takes someone literally 1 sentence of explanation and
they'll never forget what this repo is for. That being said, no cares what
I think.

I propose we put it to a vote to engineering team. Candidates include:

a) slurpee
b) KwikEMart
c) gulp-tasks

Or we just fold these changes into infra....

**Comment by OliverNChalk** [comment]
No vote, let's just do `gulp-tasks`

File: package-lock.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
       },
       "devDependencies": {
         "@proxima-oss/eslint-config": "5.0.0",
-        "@proximacapital/slurpy": "0.0.1-3",
+        "@proximacapital/gulp-tasks": "1.0.0-1",
```
Comment (OliverNChalk) [inline package-lock.json]: what's with the version number? shouldn't it be `"1.0.0"`?

**Comment by stevenpollack** [comment]
Because I screwed up with v1.0.0. and npm doesn't let you overwrite, eh?

**Comment by OliverNChalk** [comment]
You're publishing without using `np`? That's very non standard :p

File: Gulpfile.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    LintFix,
+    MDLintCheck,
+    MDLintFix,
+    Test,
+} from "@proxima-oss/gulp-tasks";
+import gulp, { TaskFunctionCallback } from "gulp";
+
+// ***** Base Tasks *****
+gulp.task("build", Build);
+gulp.task("check-node", CheckNode);
```
Comment (OliverNChalk) [inline Gulpfile.ts:24]: looks like check-node is executing inside of the `gulp-tasks` repo? I'd also be concerned that the other tasks are executing there too? Once build is fixed, please do some thorough regression testing before we consider merging

**Review by OliverNChalk — CHANGES_REQUESTED** [review]
path issues in the `gulp-tasks` pacakge

File: Gulpfile.ts (hunk 3 above)
Comment (stevenpollack) [reply Gulpfile.ts:24]: oh shit! It might be because the way the paths are coded!

**Comment by OliverNChalk** [comment]
TODO:
 - Remove the need to install `@types/mocha`, why is this happening?
   - Stop exporting/depending on them in the package
   - Set them as a dependency not a devDependency so they get packaged

File: Gulpfile.ts (hunk 3 above)
Comment (OliverNChalk) [reply Gulpfile.ts:24]: Is this resolved?

**Comment by stevenpollack** [comment]
> TODO:
> 
> * Remove the need to install `@types/mocha`, why is this happening?
>   
>   * Stop exporting/depending on them in the package
>   * Set them as a dependency not a devDependency so they get packaged

so after hours of banging my head against the wall, this is because of the `NodeJS` interface reference in the HelperFunctions file. Mocha extends this interface in its `@types/mocha` package, and the compiler incorrectly assumes we're reference mocha's interface. I've fixed the problem by not exporting the offending functions, and instead moving them to the sole task that relies on them.

File: Gulpfile.ts (hunk 3 above)
Comment (stevenpollack) [reply Gulpfile.ts:24]: yessir

**Review by OliverNChalk — APPROVED** [review]
Looks good, `check-node` is deprecated (can see Gulpfile.js history) so may want to drop from next release of `gulp-tasks`

**Pull request merged**